snap/info.go

```diff
@@ -527,9 +534,22 @@ func (app *AppInfo) ServiceFile() string {
 	return filepath.Join(dirs.SnapServicesDir, app.ServiceName())
 }
 
-// ServiceSocketFile returns the systemd socket file path for the daemon app.
-func (app *AppInfo) ServiceSocketFile() string {
-	return filepath.Join(dirs.SnapServicesDir, app.SecurityTag()+".socket")
+// ServiceSocketFiles returns the list of  systemd socket file paths for the daemon app.
+func (app *AppInfo) ServiceSocketFiles() []string {
+	socketPaths := make([]string, 0, len(app.Sockets))
+	for name := range app.Sockets {
+		socketPaths = append(socketPaths, app.SocketFile(name))
+	}
+	sort.Strings(socketPaths)
+	return socketPaths
+}
+
+// SocketFile returns the systemd socket file path for the specified app socket.
+func (app *AppInfo) SocketFile(name string) string {
```

Comment:
This one should be in SocketInfo itself, as a File method. There's an established convention in this specific file that the child types link back to the parent types so that we can use them without having to provide a trail of parents every time. For example, the first field in AppInfo is Snap, and same for PlugInfo, SlotInfo, and HookInfo. So following this strategy we want an `App *AppInfo` as the first field in SocketInfo so we can hand it around and have methods on it without having to provide the context they're under.